okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -100,6 +108,13 @@ class RealWebSocket(
   /** True if we've enqueued a close frame. No further message frames will be enqueued. */
   private var enqueuedClose = false
 
+  /**
+   * For client, suggested compression options.
+   * Actual will be defined based on the server response.
+   * For server, actual compression options for outbound stream.
+   */
+  private var options: WebSocketOptions = WebSocketOptions(compression, contextTakeover)
```

Comment:
I think the term of art for Web Sockets is extensions, not options

Reply:
These parameters are a slight simplification of the permessage-deflate spec. In particular...

contextTakeover is independent for client-sent and server-sent messages
there’s `client_max_window_bits` and `server_max_window_bits` options

We need to detect and recognize these options and reject negotiations that we cannot or do not want to implement. Most particularly, we cannot implement `client_max_window_bits` for any value other than 15.

Reply:
(we can, however, operate just fine if that option is absent).